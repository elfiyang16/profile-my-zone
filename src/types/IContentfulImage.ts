export interface IContentfulImage {
  id?: string
  title: string
  altText?: string
  file: {
    url: string
    fileName?: string
    contentType?: string
  }
}
