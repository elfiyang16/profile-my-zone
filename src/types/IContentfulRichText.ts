import { ContentfulRichTextGatsbyReference } from "gatsby-source-contentful/rich-text"

export interface IContentfulRichText<
  T extends ContentfulRichTextGatsbyReference
> {
  raw: string
  references: T[]
}
