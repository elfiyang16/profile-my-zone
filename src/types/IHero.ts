import {IContentfulImage} from "./IContentfulImage"

export interface IHero {
  colorTheme?: string
  heading1?: string
  heading2? :string
  image?:IContentfulImage
}
