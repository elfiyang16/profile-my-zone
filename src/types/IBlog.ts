import { IContentfulRichText } from "./IContentfulRichText"
import { IContentfulImage } from "./IContentfulImage"

export interface IBlog {
  id?: string
  author?: { name: string }
  slug?: string
  title: string
  description?: {
    description: string
  }
  body: IContentfulRichText
  publishDate?: string
  heroImage?: IContentfulImage
}
