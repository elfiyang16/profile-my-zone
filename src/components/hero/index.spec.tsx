import { shallow } from "enzyme"
import React from "react"
import Hero from "."
import { IHero } from "../../types/IHero"

const heroMock: IHero = {
  colorTheme: "",
  heading1: "Some title",
  heading2: "Some other title",
  image: {
    title: "The example",
    file: {
      url: "//images.ctfassets.net/example.png",
    },
  },
}

describe("<Hero />", () => {
  let wrapper: any
  beforeEach(() => {
    wrapper = shallow(<Hero {...heroMock} />)
  })

  it("should render", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
