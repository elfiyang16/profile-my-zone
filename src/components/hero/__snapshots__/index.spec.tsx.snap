// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should render 1`] = `
<div
  aria-label="Main hero"
  className="hero "
  role="region"
>
  <div
    className="hero__img-wrapper"
  >
    <img
      alt="The example"
      className="hero__img"
      src="//images.ctfassets.net/example.png"
    />
  </div>
  <section
    className="hero__header-wrapper"
  >
    <h1
      className="hero__header-primary"
    >
      Some title
    </h1>
    <h3
      className="hero__header-secondary"
    >
      Some other title
    </h3>
  </section>
</div>
`;
